src/rustdoc.md

```diff
@@ -0,0 +1,177 @@
+# The walking tour of rustdoc
+
+Rustdoc actually uses the rustc internals directly. It lives in-tree with the compiler and standard
+library. This chapter is about how it works. (A new implementation is also [under way], though).
+
+[under way]: https://github.com/steveklabnik/rustdoc
+
+Rustdoc is implemented entirely within the crate `librustdoc`. After partially compiling a crate to
+get its AST (technically the HIR map) from rustc, librustdoc performs two major steps past that to
```

Comment:
> partially compiling a crate to get its AST (technically the HIR map)

Is this a valid thing to say? The complete story of what librustdoc does is that it runs "phase 3" of the compilation process to get a `TyCtxt` and query the compiler about type information that it can't get from the source code. But saying that it compiles the crate to its HIR felt like a simpler/smoother thing to say in this context.

Reply:
What about saying something along the lines of "it runs the compiler up to the point where we have an internal representation of a crate (HIR) and the ability to query the type system (via `TyCtxt`)"?

I'm not sure referring to the `TyCtxt` as "the type system" is actually correct, but it helps give you the general idea.